client/src/service/api.ts

```diff
@@ -0,0 +1,27 @@
+export const URL = "http://localhost:3000";
+
+export enum Endpoint {
+  TEAMS = "teams",
+}
+
+type JSONResponse = {
+  status?: number;
+  data: any;
+  error: any;
+};
+
+export const request = async (endpoint: Endpoint, params?: any) => {
+  try {
+    const res = await fetch(`${URL}/${endpoint}`);
+    const { ok, status } = res;
+    const data = await res.json();
+    if (ok) {
+      return { status, data, error: false };
+    }
+    // This is an error from the API side, we get back 4xx or 5xx code
```

Comment:
Would it make sense to have "buildApiErrorMessage" and "buildClientError" (for below) methods? It would make the comments less needed, and would allow logging and/or anything else we might want to do when problems occur. 

This isn't blocking . . . the comments just raised a (very light) yellow flag.